### ISSUE TITLE
[ENH] Pass `pmdarima` and `prophet` models' `fit_params` to constructor

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#2344]
Native (not `sktime`) `pmdarima` models `ARIMA` and `AutoARIMA`, and `Prophet` (in`prophet`, not `sktime`) have some parameters which can be passed via `fit_params` (natively in `pmdarima` or `prophet`).
Users should be able to pass those parameters through the constructor of the `sktime` interface classes.
We should:

* check whether we are missing any parameters in the constructors that can be passed as `fit_params`. (if none are missing, we do not need to do the second step for that class) 
    - [ ] `Prophet`
    - [ ] `ARIMA`
    - [ ] `AutoARIMA` 
* if yes, add them to the constructor and ensure they are passed on to the interfaced model. The recipe is as below.
    - [ ] `Prophet`
    - [ ] `ARIMA`
    - [ ] `AutoARIMA` 
 
Good first issue since recipe is available.

---
Recipe for interfacing models with `fit_params` or any other `fit` args, in sktime:

* pass the parameter(s) to the constructor (the `__init__` method), using explicit parameter names and default, not `**kwargs`
* write it to `self` in the constructor
* in `_fit`, read it from `self` and pass it to the wrapped model's `fit` method
* ensure docstring is complete and defaults documented (optimally same as in wrapped estimator)

This allows to expose all parameter choices via the scikit-learn compatible interface (which relies on the constructor to expose all parameter choices).

For an example of this, you can look at `smoothing_level` in `ExponentialSmoothing`.

**Comment by Ris-Bali** [comment]
Can I work on this issue @fkiraly ?

**Comment by aiwalter** [comment]
@Ris-Bali yes pls 🙂

**Comment by fkiraly** [comment]
FYI @jasminelaw, duplicate of yout #3687.

Contributions appreciated!

**Comment by fkiraly** [comment]
@tpvasconcelos, looks reasonable - so you are saying we *are* missing parameters in `Prophet`?

a dict of params makes sense, although it tends to be unpleasant when using in tuners etc.

**Comment by tpvasconcelos** [comment]
@fkiraly Yes that seems to be the case.

As you can see, Prophet's `fit` method accepts arbitrary (and undocumented) parameters that are passed down to Stan.

https://github.com/facebook/prophet/blob/c00f6a2d72229faa6acee8292bc01e14f16f599c/python/prophet/forecaster.py#L1188-L1212
```python
    def fit(self, df, **kwargs):
        """Fit the Prophet model.
        ...

        Parameters
        ----------
        df: pd.DataFrame containing the history. Must have columns ds (date
            type) and y, the time series. If self.growth is 'logistic', then
            df must also have a column cap that specifies the capacity at
            each ds.
        kwargs: Additional arguments passed to the optimizing or sampling
            functions in Stan.
        ...
        """
```

and sktime does not pass anything as of now:

https://github.com/sktime/sktime/blob/df21a0c0275ebf28deb30efac5d469c9f0d178e3/sktime/forecasting/base/adapters/_fbprophet.py#L119-L123

I agree with your point but unfortunately, since these are undocumented and even the Prophet API accepts an arbitrary dictionary, I think we should follow suit here.

From my side, I only need this for testing purposes as this is the only way to set a random seed and select the underlying optimiser used by stan (non are deterministic but `algorithm='Newton'` yields the most consistent results). AFAIK, these would not be used for hyperparameter tuning.

I'll get started on a PR 👍

**Comment by fkiraly** [comment]
this should not be closed until we have checked all instances mentioned in the OP.

I think we can check `Prophet` off the list now.
